test/test_sparse.py

```diff
@@ -3321,22 +3329,23 @@ def test_sparse_matmul(sparse_dims, nnz, shape_a, shape_b):
             r2 = torch.sparse.mm(a, b)
             self.assertEqual(r1, r2)
 
-            a.requires_grad_(True)
-            b.requires_grad_(True)
+            if dtype in [torch.double, torch.cdouble]:
```

Comment:
Why this constraint?

Reply:
The reason is that `gradcheck` works by comparing the result of torch's autograd against the result produced by finite differencing and finite differences compute accurate derivatives only for double precision input.
This test will be refactored into smaller tests later and autograd should be checked with OpInfo-based tests.

https://github.com/pytorch/pytorch/blob/a524ee00ca8474e8d9b8bb83814316f54100acf8/torch/autograd/gradcheck.py#L646-L651